src/pyhf/infer/__init__.py

```diff
@@ -109,30 +119,38 @@ def hypotest(
     )
 
     _returns = [CLs]
+    if use_q0:
```

Comment:
what follows after here is a little bit of a mess, now with even more if-else distinctions due to the `q0` - one might consider disentangling the calculation of pvalues and CLs?